apps/requestron/openshift/dc.yml

```diff
@@ -0,0 +1,148 @@
+  metadata:
+    creationTimestamp: null
+    name: ${NAME}${SUFFIX}
+  spec:
+    replicas: 1
+    selector:
+      deploymentconfig: ${NAME}${SUFFIX}
+    strategy:
+      resources:
+        requests:
+          cpu: '100m'
+        limits:
+          cpu: '150m'
+    template:
+      metadata:
+        creationTimestamp: null
+        labels:
+          deploymentconfig: ${NAME}${SUFFIX}
+      spec:
+        containers:
+        - image: ${NAME}${VERSION}
+          name: ${NAME}
+          ports:
+          - containerPort: 8000
+            protocol: TCP
+          resources:
+            requests:
+              cpu: '100m'
+            limits:
+              cpu: '150m'
+          env:
+          - name: PRIVATE_KEY
+            valueFrom:
+              secretKeyRef:
+                name: github-private-key
+                key: PRIVATE_KEY
+          - name: WEBHOOK_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: webhook-secret
+                key: WEBHOOK_SECRET
+          - name: ZENHUB_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: zenhub-token
+                key: ZENHUB_TOKEN
+          - name: GITHUB_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: github-token
+                key: GITHUB_TOKEN
+          - name: REPO_ID
+            value: ${REPO_ID_VALUE}
+          - name: APP_ID
+            value: ${APP_ID_VALUE}
+          - name: WEBHOOK_PROXY_URL
+            value: ${WEBHOOK_PROXY_URL_VALUE}
+          - name: TZ
+            value: America/Vancouver
+    test: false
+    triggers:
+    - type: ConfigChange
+    - imageChangeParams:
+        automatic: true
+        containerNames:
+        - ${NAME}
+        from:
+          kind: ImageStreamTag
+          name: ${NAME}:${VERSION}
+      type: ImageChange
+- apiVersion: v1
+  kind: Service
+  metadata:
+    creationTimestamp: null
+    name: ${NAME}${SUFFIX}
+  spec:
+    ports:
+    - name: api
+      port: 8000
+      protocol: TCP
+      targetPort: 8000
+    selector:
+      deploymentconfig: ${NAME}${SUFFIX}
+- apiVersion: route.openshift.io/v1
+  kind: Route
+  metadata:
+    creationTimestamp: null
+    name: ${NAME}${SUFFIX}
+  spec:
+    host:
+    port:
+      targetPort: api
+    tls:
+      insecureEdgeTerminationPolicy: Redirect
+      termination: edge
+    to:
+      kind: Service
+      name: ${NAME}${SUFFIX}
+      weight: 100
+    wildcardPolicy: None
+parameters:
+- description: A name used for all objects
+  displayName: Name
+  name: NAME
+  required: true
+  value: requestron
+- description: A name suffix used for all objects
+  displayName: Suffix
+  name: SUFFIX
+  required: false
+- description: A version used for the image tags
+  displayName: version
+  name: VERSION
+  required: true
+  value: v1.0.0
+- description: github ID of the request repo
+  displayName: repo id value
+  name: REPO_ID_VALUE
+  required: true
+  value: "219808631"
```

Comment:
this ideally could be in the config.js file in .pipeline/  Try to keep deployment and build configs parameterized and general.